File: src/app/types/passphrase.ts
export interface PassphraseConfig {
    dict_english: boolean | false;
    dict_spanish: boolean | false;
    words: number;
    min_length: number;
    max_length: number;
    separator: string | ' ';
    salt: boolean | false;
}

export interface PassphraseResult {
    passphrase: string;
    strength: number | null;
}

File: src-tauri/src/types/passphrase.rs
use serde::{Deserialize, Serialize};

#[derive(Deserialize)]
pub struct PassphraseConfig {
    pub dict_english: bool,
    pub dict_spanish: bool,
    pub words: usize,
    pub min_length: usize,
    pub max_length: usize,
    pub separator: String,
    pub salt: bool,
}

#[derive(Serialize)]
pub struct PassphraseResult {
    pub passphrase: String,
    pub strength: f64,
}